### PR TITLE
Update NEWS for cvc5 1.0.

### PR DESCRIPTION
**Pull request by mpreiner**
Co-authored-by: Gereon Kremer <nafur42@gmail.com>
Co-authored-by: Andrew Reynolds <andrew.j.reynolds@gmail.com>
Co-authored-by: mudathirmahgoub <mudathirmahgoub@gmail.com>
Co-authored-by: Andres Noetzli <andres.noetzli@gmail.com>

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+* *New Java language bindings*
+
+* *Theory of Bags (Multisets)*
+  - TODO: describe Bags
+
+* *Theory of Sequences*
+  - A new parametric theory of sequences whose syntax is compatible with the
+    syntax for sequences used by Z3.
+  - A new array-inspired procedure
```
Comment (ajreynol) [inline NEWS.md:22]:
```suggestion
  - A new array-inspired procedure (option `--seq-array`).
```

File: NEWS.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+* *Arrays*
+  - Added support for an `eqrange` predicate. `(eqrange a b i j)` is true
+    if arrays `a` and `b` are equal on all indices within indices `i` and `j`.
+
+* *Bit-vectors*
+  - New bit-vector solver with CaDiCaL as default SAT back-end.
+  - New approach for solving bit-vectors as integers
+
+* *Datatypes*
+  - Updaters: TODO
```
Comment (ajreynol) [inline NEWS.md:36]:
```suggestion
  - Support for generic datatype updaters `((_ update s) t u)` which replaces the field specified by selector `s` of `t` by the value `u`. 
```

File: NEWS.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+* *Integers*
+  - Support for an integer operator `(_ iand n)` that returns the bitwise `and`
+    of two integers, seen as integers modulo n.
+  - Support for an integer operator `int.pow2`, used as `(int.pow2 x)` which
+    represents 2 to the power of x.
+
+* *Quantifiers*
+  - Get list of relevant Instantiations
```
Comment (ajreynol) [inline NEWS.md:45]:
```suggestion
```

Comment (ajreynol) [reply NEWS.md:45]: On second thought, our API is not ideal for this yet, let's not mention it.

File: NEWS.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  - Improved support for enumerative instantiation
+  - SyGuS-based quantifier instantiation
+
+* *Strings*
+  - Support for `str.indexof_re(s, r, n)`, which returns the index of the first
```
Comment (ajreynol) [inline NEWS.md:50]:
```suggestion
  - Improved performance using new techniques, including model-based reductions, eager context-dependent simplification, and equality-based conflict finding.
  - Support for `str.indexof_re(s, r, n)`, which returns the index of the first
```

File: NEWS.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+* *Theory of Bags (Multisets)*
+  - TODO: describe Bags
+
+* *Theory of Sequences*
+  - A new parametric theory of sequences whose syntax is compatible with the
+    syntax for sequences used by Z3.
+  - A new array-inspired procedure
+
+* *Arithmetic*
+  - New nonlinear arithmetic solver based on cylindrical algebraic coverings.
```
Comment (nafur) [inline NEWS.md:25]:
```suggestion
* *Arithmetic*
  - Nonlinear real arithmetic is now solved using a new solver based on cylindrical
    algebraic coverings. Includes full support for non-rational models and a number
    of options `--nl-cov-*` for, e.g., different projection operators, Lazard's
    lifting or variable elimination.
```

File: NEWS.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+* Interactive shell: the GPL-licensed Readline library has been replaced the
+  BSD-licensed Editline. Compiling with `--best` now enables Editline, instead
+  of Readline. Without selecting optional GPL components, Editline-enabled CVC4
+  builds will be BSD licensed.
+* The semantics for division and remainder operators in the CVC language now
+  correspond to SMT-LIB 2.6 semantics (i.e. a division by zero or a zero
+  modulus results in a constant value, instead of an uninterpreted one). As a
+  result the option `--bv-div-zero-const` has been removed. Similarly, when no
+  language is set, the API semantics now correspond to the SMT-LIB 2.6
+  semantics.
```
Comment (ajreynol) [inline NEWS.md:135]:
```suggestion
```

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+    the field specified by selector `s` of `t` by the value `u`.
+
+* *Integers*
+  - Support for an integer operator `(_ iand n)` that returns the bitwise `and`
+    of two integers, seen as integers modulo n.
+  - Support for an integer operator `int.pow2`, used as `(int.pow2 x)` which
+    represents 2 to the power of x.
+
+* *Quantifiers*
+  - Improved support for enumerative instantiation
```
Comment (ajreynol) [inline NEWS.md:49]:
```suggestion
  - Improved support for enumerative instantiation (option `--enum-inst`).
```

File: NEWS.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+  - SyGuS-based quantifier instantiation
```
Comment (ajreynol) [inline NEWS.md:50]:
```suggestion
  - SyGuS-based quantifier instantiation (option `--sygus-inst`).
```

File: NEWS.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+* *Unsat cores*
+  - Production modes based on the new proof infrastructure
+    (`--unsat-cores-mode=sat-proof`) and on the solving-under-assumption
+    feature of Minisat (`--unsat-cores-mode=assumptions`). The mode based on
+    SAT assumptions + preprocessing proofs is the new default.
+  - Computing minimal unsat cores (option `--minimal-unsat-cores`).
+
+* *Blocking Models*
+  - TODO
```
Comment (ajreynol) [inline NEWS.md:118]:
```suggestion
  - The API method `blockModels` can be used to block the current model based a mode (`block-model` in SMT-LIB).
  - The API method `blockModelValues` can be used to block the current model for a provided set of terms (`block-model-values` in SMT-LIB).
```

File: NEWS.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+* *Model Cores*
+  - TODO
```
Comment (ajreynol) [inline NEWS.md:121]:
```suggestion
  - The API method `isModelCoreSymbol` can be used to query whether the value for a symbol was critical to whether the model satisfies the current set of assertions.
  - Models can be limited to show only model core symbols (option `--model-cores`).
```

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+* *Streamlined C++ API*
+
+* *Two new Python language bindings*
+  - Base module: Feature complete with C++ API
+  - Pythonic module: More pythonic bindings based on the base module
+
+* *New Java language bindings*
+
+* *Theory of Bags (Multisets)*
+  - TODO: describe Bags
```
Comment (mudathirmahgoub) [inline NEWS.md:24]:
```suggestion
  A new theory of bags which are sets that allow duplicates. It supports basic operators like union disjoint, union max, intersection, difference subtract, difference remove, duplicate removal, and multiplicity of an element in a bag. 
```

Comment (ajreynol) [reply NEWS.md:24]: Reworded to "collections" to be generic (they are not sets).

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+    Ozdemir A., Preiner M., Reynolds A., Sheng Y., Tinelli C., and Zohar Y.,
+    TACAS 2022.
+
+**New Features**
+
+* *Streamlined C++ API*
+
+* *Two new Python language bindings*
+  - Base module: Feature complete with C++ API
+  - Pythonic module: More pythonic bindings based on the base module
```
Comment (4tXJ7f) [inline NEWS.md:19]:
```suggestion
  - Pythonic module: A pythonic wrapper around the base module
```

File: NEWS.md [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+**Changes**
+
+* CaDiCaL and SymFPU are now required dependencies.
+* Options have been extensively refactored, please refer to the cvc5
+  documentation for further information.
+* Removed support for the CVC language.
+* SyGuS: Removed support for SyGuS-IF 1.0.
+* Removed support for the (non-standard) `define` command.
+* Removed Java and Python bindings for the legacy API.
```
Comment (4tXJ7f) [inline NEWS.md:140]:
```suggestion
* Removed the legacy API along with the Java and Python bindings for it.
```

File: NEWS.md [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+* Interactive shell: the GPL-licensed Readline library has been replaced the
+  BSD-licensed Editline. Compiling with `--best` now enables Editline, instead
+  of Readline. Without selecting optional GPL components, Editline-enabled CVC4
+  builds will be BSD licensed.
+* The `competition` build type includes the dependencies used for SMT-COMP by
+  default. Note that this makes this build type produce GPL-licensed binaries.
+* Bit-vector operator bvxnor was previously mistakenly marked as
```
Comment (4tXJ7f) [inline NEWS.md:147]:
```suggestion
* Bit-vector operator `bvxnor` was previously mistakenly marked as
```

File: NEWS.md [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+**System Description**
+  * *cvc5: A Versatile and Industrial-Strength SMT Solver.*  
+    Barbosa H., Barrett C., Brain M., Kremer G., Lachnitt H.,
+    Mann M., Mohamed A., Mohamed M., Niemetz A., Nötzli A.,
+    Ozdemir A., Preiner M., Reynolds A., Sheng Y., Tinelli C., and Zohar Y.,
+    TACAS 2022.
+
+**New Features**
+
+* *Streamlined C++ API*
```
Comment (4tXJ7f) [inline NEWS.md:15]:
```suggestion
* *Streamlined C++ API*
  Please refer to the documentation for details.
```

File: NEWS.md [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+* *Theory of Sequences*
+  - A new parametric theory of sequences whose syntax is compatible with the
+    syntax for sequences used by Z3.
+  - A new array-inspired procedure (option `--seq-array`).
+
+* *Arithmetic*
+  - Nonlinear real arithmetic is now solved using a new solver based on
+    cylindrical algebraic coverings. Includes full support for non-rational
+    models and a number of options `--nl-cov-*` for, e.g., different projection
+    operators, Lazard's lifting or variable elimination.
```
Comment (4tXJ7f) [inline NEWS.md:35]: @nafur We should probably also mention the new optional dependencies.

**Review by 4tXJ7f — COMMENTED** [review]
Some suggestions.

File: NEWS.md (hunk 16 above)
Comment (nafur) [inline NEWS.md:35]:
```suggestion
* *Arithmetic*
  - Nonlinear real arithmetic is now solved using a new solver based on
    cylindrical algebraic coverings. Includes full support for non-rational
    models and a number of options `--nl-cov-*` for, e.g., different projection
    operators, Lazard's lifting or variable elimination.
    The new solver requires the libpoly library, and Lazard's lifting
    additionally requires CoCoALib.
```

File: NEWS.md [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+* *Strings*
+  - Improved performance using new techniques, including model-based
+    reductions, eager context-dependent simplification, and equality-based
+    conflict finding.
+  - Support for `str.indexof_re(s, r, n)`, which returns the index of the first
+    occurrence of a regular expression `r` in a string `s` after index `n` or
+    -1 if `r` does not match a substring after `n`.
+
+* *Proofs*
+  * When used after an unsatisfiable response to checkSat, getProof
```
[the remark this answers is not in the document]
Comment (ajreynol) [reply NEWS.md:68]: @nafur I'm not sure how would be best to link to the general proof documentation here.  What should we put?

[the remark this answers is not in the document]
Comment (nafur) [reply NEWS.md:71]: We could use https://cvc5.github.io/docs/cvc5-1.0.0/proofs/proofs.html, being slightly optimistic that everything just works :-)

[the remark this answers is not in the document]
Comment (mpreiner) [reply NEWS.md:73]: Added the link.

File: NEWS.md [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+* Bit-vector operator `bvxnor` was previously mistakenly marked as
+  left-assoicative in SMT-LIB. This has recently been corrected in SMT-LIB. We
+  now restrict `bvxnor` to only allow two operands in order to avoid confusion
+  about the semantics, since the behavior of n-ary operands to `bvxnor` is now
+  undefined in SMT-LIB.
+* SMT-LIB output for `get-model` command now conforms with the standard,
+  and does *not* begin with the keyword `model`. The output
+  is the same as before, only with this word removed from the beginning.
+* Building with Python 2 is now deprecated.
+* Removed support for redundant logics `ALL_SUPPORTED` and `QF_ALL_SUPPORTED`,
```
Comment (ajreynol) [inline NEWS.md:161]:
```suggestion
* The SMT-LIB syntax for some extensions has been changed. Notably, set operators are now prefixed by `set.`. For example, `union` is now written `set.union`.
* Removed support for redundant logics `ALL_SUPPORTED` and `QF_ALL_SUPPORTED`,
```

File: NEWS.md [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+  Please refer to the documentation for details.
+
+* *Two new Python language bindings*
+  - Base module: Feature complete with C++ API
+  - Pythonic module: A pythonic wrapper around the base module
+
+* *New Java language bindings*
+
+* *Theory of Bags (Multisets)*
+  A new theory of bags, which are collections that allow duplicates. It supports basic operators like union disjoint, union max, intersection, difference subtract, difference remove, duplicate removal, and multiplicity of an element in a bag.
```
Comment (ajreynol) [inline NEWS.md:25]:
```suggestion
  - A new theory of bags, which are collections that allow duplicates. It supports basic operators like union disjoint, union max, intersection, difference subtract, difference remove, duplicate removal, and multiplicity of an element in a bag. 
```

File: NEWS.md [hunk 20]
```diff
@@ -0,0 +1,163 @@
...
+    The new solver requires the libpoly library, and Lazard's lifting
+    additionally requires CoCoALib.
+
+* *Arrays*
+  - Added support for an `eqrange` predicate. `(eqrange a b i j)` is true
+    if arrays `a` and `b` are equal on all indices within indices `i` and `j`.
+
+* *Bit-vectors*
+  - New bit-vector solver with CaDiCaL as default SAT back-end.
+  - New approach for solving bit-vectors as integers
```
Comment (ajreynol) [inline NEWS.md:46]:
```suggestion
  - New approach for solving bit-vectors as integers (option `--solve-bv-as-int`).
```

File: NEWS.md [hunk 21]
```diff
@@ -0,0 +1,163 @@
...
+    occurrence of a regular expression `r` in a string `s` after index `n` or
+    -1 if `r` does not match a substring after `n`.
+
+* *Proofs*
+  * Documentation available at:
+    https://cvc5.github.io/docs/cvc5-1.0.0/proofs/proofs.html
+  * When used after an unsatisfiable response to checkSat, getProof
+    returns a representation of the refutation proof for the current set of
+    assertions (get-proof in SMT-LIB).
+
```
Comment (ajreynol) [inline NEWS.md:76]:
```suggestion
  * Preliminary support for translations to external proof formats LFSC and Alethe.

```

File: NEWS.md [hunk 22]
```diff
@@ -0,0 +1,163 @@
...
+* *Difficulty Estimation*
+  - The API method getDifficulty returns a map from asserted formulas to
+    integers which estimates how much that formula contributed to the
+    difficulty of the overall problem (get-difficulty in SMT-LIB).
+
+* *Learned Literals*
+  - The API method  getLearnedLiterals gets a list of literals that are
```
Comment (ajreynol) [inline NEWS.md:83]:
```suggestion
  - The API method getLearnedLiterals gets a list of literals that are
```

File: NEWS.md [hunk 23]
```diff
@@ -0,0 +1,163 @@
...
+    synthesis conjecture via the API. The method checkSynthNext finds the
+    next SyGuS solution to the current set of SyGuS constraints
+    (check-synth-next in SyGuS IF).
+  - getAbductNext finds the next abduct for the current set of
+    assertions and provided goal (get-abduct-next in SMT-LIB).
+  - getInterpolantNext finds the next interpolant for the current set of
+    assertions and provided goal (get-interpolant-next in SMT-LIB).
+
+* *Pool-based Quantifier Instantiation*
+  - declarePool declares symbol sets of terms called pools (declare-pool in
```
Comment (ajreynol) [inline NEWS.md:109]:
```suggestion
  - The API method declarePool declares symbol sets of terms called pools (declare-pool in
```

File: NEWS.md [hunk 24]
```diff
@@ -0,0 +1,163 @@
...
+* Bit-vector operator `bvxnor` was previously mistakenly marked as
+  left-assoicative in SMT-LIB. This has recently been corrected in SMT-LIB. We
+  now restrict `bvxnor` to only allow two operands in order to avoid confusion
+  about the semantics, since the behavior of n-ary operands to `bvxnor` is now
+  undefined in SMT-LIB.
+* SMT-LIB output for `get-model` command now conforms with the standard,
+  and does *not* begin with the keyword `model`. The output
+  is the same as before, only with this word removed from the beginning.
+* Building with Python 2 is now deprecated.
+* The SMT-LIB syntax for some extensions has been changed. Notably, set operators are now prefixed by `set.`. For example, `union` is now written `set.union`.
```
Comment (ajreynol) [inline NEWS.md:161]:
```suggestion
* The SMT-LIB syntax for some extensions has been changed. Notably, set operators are now prefixed by `set.`, and relations by `rel.`. For example, `union` is now written `set.union`.
```

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: NEWS.md [hunk 25]
```diff
@@ -111,8 +116,8 @@ cvc5 1.0
     assertions and provided goal (get-interpolant-next in SMT-LIB).
 
 * *Pool-based Quantifier Instantiation*
-  - The API method declarePool declares symbol sets of terms called pools (declare-pool in
-    SMT-LIB).
+  - The API method declarePool declares symbol sets of terms called pools
+    (declare-pool in SMT-LIB).
```
Comment (nafur) [inline NEWS.md:120]:
```suggestion
    (`declare-pool` in SMT-LIB).
```

**Review by ajreynol — APPROVED** [review]

**Pull request merged**